src/data/nat/pow.lean

```diff
@@ -113,6 +113,27 @@ by { rw [sq, sq], exact nat.mul_self_sub_mul_self_eq a b }
 
 alias sq_sub_sq ← pow_two_sub_pow_two
 
+theorem cauchy_induction (k : ℕ) {P : ℕ → Prop} (hk : 1 < k)
```

Comment:
You can generalize this by changing `h2` to `∀ (n : ℕ), P n → P (f n)` where `f` satisfies `∀ x > m, x < f x`. If you want to, you can then prove standard Cauchy induction from this more general principle.

Reply:
In fact, you can prove the even more general principle: if a proposition is true for infinitely many values / arbitrarily large values, and `P (n + 1) → P n`, then it's true for all values.

Reply:
In fact [nat.decreasing_induction](https://leanprover-community.github.io/mathlib_docs/data/nat/basic.html#nat.decreasing_induction) already shows that (given `P (n + 1) → P n`) if for every n there is m ≥ n such that `P m` (*), then `P n` holds for all `n`.

(*) The condition is also equivalent to `¬ bdd_above {m | P m}` and `{m | P m}.infinite`, but mathlib is missing the equivalence of the latter two (only has one direction of the implication as [set.infinite_of_not_bdd_above](https://leanprover-community.github.io/mathlib_docs/data/set/finite.html#set.infinite_of_not_bdd_above)).